src/IntercomChannel.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace FtwSoft\NotificationChannels\Intercom;
+
+use Intercom\IntercomClient;
+use Illuminate\Notifications\Notification;
+use GuzzleHttp\Exception\BadResponseException;
+use FtwSoft\NotificationChannels\Intercom\Exceptions\RequestException;
+use FtwSoft\NotificationChannels\Intercom\Contracts\IntercomNotification;
+use FtwSoft\NotificationChannels\Intercom\Exceptions\InvalidArgumentException;
+use FtwSoft\NotificationChannels\Intercom\Exceptions\MessageIsNotCompleteException;
+
+/**
+ * Class IntercomNotificationChannel.
+ */
+class IntercomChannel
+{
+    /**
+     * @var \Intercom\IntercomClient
+     */
+    private $client;
+
+    /**
+     * IntercomNotificationChannel constructor.
+     *
+     * @param \Intercom\IntercomClient $client
+     */
+    public function __construct(IntercomClient $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Send the given notification via Intercom API.
+     *
+     * @param mixed                                  $notifiable
+     * @param \Illuminate\Notifications\Notification $notification
+     *
+     * @throws \FtwSoft\NotificationChannels\Intercom\Exceptions\RequestException              When server responses with a bad HTTP
+     *                                                                                         code
+     * @throws \FtwSoft\NotificationChannels\Intercom\Exceptions\MessageIsNotCompleteException When message is not
+     *                                                                                         filled correctly
+     * @throws \GuzzleHttp\Exception\GuzzleException                                           Other Guzzle uncatched exceptions
+     *
+     * @return void
+     *
+     * @see https://developers.intercom.com/intercom-api-reference/reference#admin-initiated-conversation
+     */
+    public function send($notifiable, Notification $notification): void
+    {
+        try {
+            $this->sendNotification($notifiable, $notification);
+        } catch (BadResponseException $exception) {
+            throw new RequestException($exception, $exception->getMessage(), $exception->getCode());
+        }
+    }
+
+    /**
+     * @return \Intercom\IntercomClient
+     */
+    public function getClient(): IntercomClient
+    {
+        return $this->client;
+    }
+
+    /**
+     * @param mixed        $notifiable
+     * @param Notification $notification
+     *
+     * @throws MessageIsNotCompleteException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    protected function sendNotification($notifiable, Notification $notification): void
+    {
+        if (false === $notification instanceof IntercomNotification) {
```

Comment:
I don't think we should require the notification to implement a contract. I think we shouldn't throw and exception in this case